### PR TITLE
Testing: Fix failing Writing Flow test

### PR DESCRIPTION
**Pull request by aduth**
Regression of #9977

This pull request seeks to resolve a failing E2E test, failed to have been updated in #9977.

**Testing instructions**

Verify E2E tests pass:

```
npm run test-e2e
```

### CODE REVIEW

File: test/e2e/specs/writing-flow.test.js [hunk 1]
```diff
@@ -149,13 +149,11 @@ describe( 'adding blocks', () => {
 		await pressWithModifier( META_KEY, 'b' );
 		expect( await getEditedPostContent() ).toMatchSnapshot();
 
-		// When returning to Visual mode, backspace in selected block should
-		// reset to an unmodified default block.
+		// Reset.
```
Comment (tofumatt) [inline test/e2e/specs/writing-flow.test.js]: Concise. (The old comment seemed better. 🤔)

File: test/e2e/specs/writing-flow.test.js [hunk 2]
```diff
@@ -149,13 +149,11 @@ describe( 'adding blocks', () => {
...
 		await page.keyboard.press( 'Backspace' );
 
 		// Ensure no data-mce-selected. Notably, this can occur when content
 		// is saved while typing within an inline boundary.
-		await clickBlockAppender();
```
Comment (tofumatt) [inline test/e2e/specs/writing-flow.test.js]: Why the change? I believe you, but it's hard to tell because there isn't accompanying code in the PR.

**Review by tofumatt — COMMENTED** [review]
Why the comment-removal?

File: test/e2e/specs/writing-flow.test.js (hunk 1 above)
Comment (aduth) [reply test/e2e/specs/writing-flow.test.js]: the old comment was wrong (outdated). we don't switch modes

Comment (tofumatt) [reply test/e2e/specs/writing-flow.test.js]: It can probably just be removed then. Or expanded. "Remove text to reset the block." or something. 🤷‍♂️

File: test/e2e/specs/writing-flow.test.js (hunk 2 above)
Comment (aduth) [reply test/e2e/specs/writing-flow.test.js]: Before #9977, it was possible to remove all blocks. The default appender was shown.

After #9977, removing all blocks causes an empty default block to be inserted, at which point the default appender is not present on the page (hence the original error).

Comment (tofumatt) [reply test/e2e/specs/writing-flow.test.js]: 👍

**Review by tofumatt — APPROVED** [review]
more-red-than-green
